src/problems/shinny_sam_sudoku.clj

```diff
@@ -0,0 +1,150 @@
+(ns problems.shinny-sam-sudoku)
+
+(defrecord State [board curr-position])
+
+;check row/column for duplicates
+(defn duplicates? [coll]
+  (not (apply distinct? (filter #(> % 0) coll))))
+
+;example to test all the other functions
+
+
+;check if a value/element is in a collection
+(defn in? [coll value]
+  (= true (some #(= value %) coll)))
+
+;return a row
+(defn row-seq [board y]
+  (nth board y))
+
+;return a column
+(defn col-seq [board x]
+  (for [y board]
+    (last(take (+ x 1) y))))
+
+;check if a row, column or 3x3 is legal
+(defn legal? [coll]
+  (and (not (duplicates? coll))))
+
+
+;the range of number(1-9) to put into the board
+(def numBank (range 1 10))
+
+;put numbers into the current position of the board
+(defn add [board position number numBank]
+  (let [num (nth numBank (- number 1))]
+    (assoc-in board position num)))
+
+
+;update the current position of the board (don't know if I can call the function and return a new position somewhere else)
+(defn updatePos [state]
+  (let [x (first (:curr-position state))
+        y (second (:curr-position state))]
+
+    (if (= y 8)
+      [(+ x 1) (- y 8)]
+      [(+ x 0) (+ y 1)])))
+
+
+(def start-state (->State [       [4 0 5 2 6 9 7 8 1]
+                                  [6 8 2 5 7 1 4 9 3]
+                                  [1 9 7 8 3 4 5 6 2]
+
+                                  [8 2 6 1 9 5 3 4 7]
+                                  [3 7 4 6 8 2 9 1 5]
+                                  [0 5 0 7 4 3 6 2 8]
+
+                                  [5 1 9 3 2 6 8 7 4]
+                                  [2 4 8 9 5 7 1 3 6]
+                                  [7 6 3 4 1 8 2 5 9]] [0, 0]))
+
+
+
+
+;go through the number 1-9 to put in the current position and produce different childrens of States of the board(haven't implement the legal function yet)
```

Comment:
I can understand why this was once here, but you want to delete things like this before you "turn things in" (or generate pull requests).